Sources/JExtractSwiftLib/JNI/JNISwift2JavaGenerator+JavaTranslation.swift

```diff
@@ -0,0 +1,133 @@
+
+    translatedDecls[decl] = translated
+    return translated
+  }
+
+  struct JavaTranslation {
+    func translate(_ decl: ImportedFunc) -> TranslatedFunctionDecl {
+      let translatedFunctionSignature = translate(functionSignature: decl.functionSignature)
+      let parentName = decl.parentType?.asNominalType?.nominalTypeDecl.qualifiedName
+
+      return TranslatedFunctionDecl(
+        name: decl.name,
+        parentName: parentName,
+        translatedFunctionSignature: translatedFunctionSignature
+      )
+    }
+
+    func translate(functionSignature: SwiftFunctionSignature, isInitializer: Bool = false) -> TranslatedFunctionSignature {
+      let parameters = functionSignature.parameters.enumerated().map { idx, param in
+        let parameterName = param.parameterName ?? "arg\(idx))"
+        return translate(swiftParam: param, parameterName: parameterName)
+      }
+
+      return TranslatedFunctionSignature(
+        parameters: parameters,
+        resultType: translate(swiftType: functionSignature.result.type)
+      )
+    }
+
+    func translate(swiftParam: SwiftParameter, parameterName: String) -> JavaParameter {
+      return JavaParameter(
+        name: parameterName,
+        type: translate(swiftType: swiftParam.type)
+      )
+    }
+
+    func translate(swiftType: SwiftType) -> JavaType {
+      switch swiftType {
+      case .nominal(let nominalType):
+        if let knownType = nominalType.nominalTypeDecl.knownStandardLibraryType {
+          guard let javaType = translate(standardLibraryType: knownType) else {
+            fatalError("unsupported known type: \(knownType)")
+          }
+          return javaType
+        }
+
+        return .class(package: nil, name: nominalType.nominalTypeDecl.name)
+
+      case .tuple([]):
+        return .void
+
+      case .metatype, .optional, .tuple, .function:
+        fatalError("unsupported type: \(self)")
+      }
+    }
+
+    func translate(standardLibraryType: SwiftStandardLibraryTypeKind) -> JavaType? {
+      switch standardLibraryType {
+      case .bool: .boolean
+      case .int8: .byte
+      case .uint16: .char
+      case .int16: .short
+      case .int32: .int
+      case .int64: .long
+      case .float: .float
+      case .double: .double
+      case .void: .void
+      case .string: .javaLangString
+      case .int, .uint, .uint8, .uint32, .uint64,
+          .unsafeRawPointer, .unsafeMutableRawPointer,
+          .unsafePointer, .unsafeMutablePointer,
+          .unsafeRawBufferPointer, .unsafeMutableRawBufferPointer,
+          .unsafeBufferPointer, .unsafeMutableBufferPointer:
+        nil
+      }
+    }
+  }
+
+  struct TranslatedFunctionDecl {
+    /// Java function name
+    let name: String
+
+    /// The name of the parent scope this function is declared in (or nil if global)
+    let parentName: String?
+
+    /// Function signature
+    let translatedFunctionSignature: TranslatedFunctionSignature
+  }
+
+  struct JavaParameter {
```

Comment:
we have this type already, can we reuse the one from `FFMSwift2JavaGenerator+JavaTranslation.swift`? You can just move it out from the FFM file into some common place